tests/Browser/PilotPassportChallengeTest.php

```diff
@@ -0,0 +1,159 @@
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Enrollments')
+                    ->assertSee('Select a path below and enroll to get started.');
+        });
+    }
+
+    public function test_enroll(): void {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+            ->clickLink('Enrollments')
+            ->press('@enroll_1')
+            ->assertSee('You are now enrolled in the ZTL Pilot Passport program!');
+        });
+    }
+
+    public function test_passport_book_view(): void {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Passport Book')
+                    ->assertSee('Visited on: ');
+        });
+    }
+
+    public function test_achievements_view(): void {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Achievements')
+                    ->assertSee('Completed on: ');
+        });
+    }
+
+    public function test_settings_view(): void {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->assertSee('ZTL understands that some pilots may want to participate');
+        });
+    }
+
+    public function test_change_privacy_settings(): void {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->radio('@privacy', '2')
+                    ->press('Save Settings')
+                    ->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->assertRadioSelected('@privacy', '2');
+        });
+    }
+
+    public function test_disenroll_purge(): void {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->press('@purge_data')
+                    ->whenAvailable('.modal', function (Browser $modal) {
+                        $modal->type('@confirm', 'confirm - purge all')
+                            ->press('Continue');
+                    })
+                    ->waitFor('.navbar', 30)
+                    ->assertSee('You have been successfully logged out');
+        });
+    }
+
+    public function test_purge_data(): void {
+        $cid = Config::get('vatsim.auth_dev_credential');
+        $pilot = RealopsPilot::find($cid);
+        $enrollment = PilotPassportEnrollment::where('cid', $cid)->first();
+        $achievement = PilotPassportAward::where('cid', $cid)->first();
+        $log = PilotPassportLog::where('cid', $cid)->first();
+        if ($pilot || $enrollment || $achievement || $log) {
+            $this->assertTrue(false);
+        }
+        $this->assertTrue(true);
```

Comment:
We can replace this with:

`$this->assertFalse($pilot || $enrollment || $achievement || $log)`

or if you want to be more descriptive of what's being tested:

```php
$data_remaining = $pilot || $enrollment || $achievement || $log;
$this->assertFalse($data_remaining);
```

 I think I'd also like this to be included in `test_disenroll_purge()` instead of creating a new test. I don't like tests depending on other tests to run for a pass. Especially if we implement parallel testing, we cannot guarantee that tests will run in a specific order every time which could lead to random errors (and tough to debug once we forget about this code)

Reply:
Updated the first test per your comment. Regarding `test_disenroll_purge()` if I add this as another assert, it will give that test 2 asserts which you said is bad practice. This test is specifically checking front-end sequence and I want the assertSee to verify that the user is transported back to the landing page after the logout sequence (assertSee checks for the logout message).

Reply:
Hmm I might go back on what I said there (sorry if that's causing more work). I guess I shouldn't have said one assert, but rather only test one thing (we don't want to be testing more than one component in a test). However, we can't have tests depend on each other because that will eventually cause a weird test failure that we will struggle to track down. I'd rather have 2 associated asserts in a test than 2 separate tests that depend on each other. @c0repwn3r what are your thoughts here?

Reply:
@iccowan Ok, well if this is the case then I need to rebuild most of the tests. I'm a bit confused on this one though, because I feel like we'd be testing the same thing twice (were the database entries purged or not). We'd just need to test for this once, right? The test that you are referencing doesn't ask that question - it is addressing the flow of the UI.